config/backend-config/workspace-config.go

```diff
@@ -46,5 +46,19 @@ func (workspaceConfig *WorkspaceConfig) GetBackendConfig() (SourcesT, bool) {
 		logger.Error("Errored while parsing request", err, string(respBody), resp.StatusCode)
 		return SourcesT{}, false
 	}
-	return sourcesJSON, true
+	var modifiedSources SourcesT
```

Comment:
We should have this filtering logic in config-backend instead of here. Like we currently do for useNativeSDK flag.